lib/helpers/attachment.js

```diff
@@ -0,0 +1,108 @@
+const { Readable } = require('stream');
+const mime = require('mime-types');
+const request = require('request');
+const requestPromise = require('request-promise');
+
+const client = require('elasticio-rest-node')();
+
+const MetaLoader = require('../helpers/metaLoader');
+
+
+async function downloadFile(url, headers) {
```

Comment:
To avoid the implementation of common handlers for attachments, you can use https://www.npmjs.com/package/@elastic.io/component-commons-library
Currently, we are working on the documentation. Now, you can inspect a source code to find available features for attachments, for example
https://github.com/elasticio/component-commons-library/blob/master/lib/attachment/AttachmentProcessor.ts

Reply:
We already use `elasticio-rest-node` to get (get/put) urls, it's very convenient. And we will try to use your attachment lib for the SugarCRM, we have similar functionality there. Thanks for the tip.